ratis-netty/src/main/java/org/apache/ratis/netty/server/DataStreamRequestDecoder.java

```diff
@@ -32,28 +31,21 @@ public DataStreamRequestDecoder() {
   }
 
   @Override
-  protected void decode(ChannelHandlerContext channelHandlerContext,
-                        ByteBuf byteBuf,
-                        List<Object> list) throws Exception {
+  protected void decode(ChannelHandlerContext channelHandlerContext, ByteBuf byteBuf, List<Object> list) {
     if(byteBuf.readableBytes() >= 24){
       long streamId = byteBuf.readLong();
       long dataOffset = byteBuf.readLong();
       long dataLength = byteBuf.readLong();
       if(byteBuf.readableBytes() >= dataLength){
         ByteBuf bf = byteBuf.slice(byteBuf.readerIndex(), (int)dataLength);
         bf.retain();
-        DataStreamRequestServer req = new DataStreamRequestServer(streamId,
-            dataOffset,
-            bf);
+        final DataStreamRequestByteBuf req = new DataStreamRequestByteBuf(streamId, dataOffset, bf);
         byteBuf.readerIndex(byteBuf.readerIndex() + (int)dataLength);
         byteBuf.markReaderIndex();
         list.add(req);
       } else {
         byteBuf.resetReaderIndex();
```

Comment:
if `byteBuf.readableBytes() >= dataLength` is false, do we need to print err log ?

Reply:
This may be a good suggestion but is not related to this issue.  This is a refactoring but not try to change anything to the code.